File: Graph/Leetcode/Medium/Provinces/Provinces.cpp
// https://leetcode.com/problems/number-of-provinces/
#include<bits/stdc++.h>
using namespace std;
void dfs(int start, vector<vector<int>> &isConnected, vector<bool> &visited)
{
    visited[start] = true;
    for (int i = 0; i < isConnected.size(); i++)
    {
        if (isConnected[start][i] == 1 && !visited[i])
        {
            dfs(i, isConnected, visited);
        }
    }
}

int findCircleNum(vector<vector<int>> &isConnected)
{
    vector<bool> visited(isConnected.size(), 0);
    int count = 0;
    for (int i = 0; i < isConnected.size(); i++)
    {
        if (!visited[i])
        {
            count++;
            dfs(i, isConnected, visited);
        }
    }
    return count;
}
int main(void) {
    
    return 0;
}

File: Graph/Leetcode/Medium/Provinces/Provinces.java
// https://leetcode.com/problems/number-of-provinces/
public class Provinces {
    void dfs(int[][] isConnected, int start, boolean[] visited) {
        visited[start] = true;
        for (int i = 0; i < isConnected.length; i++) {
            if (isConnected[start][i] == 1 && !visited[i]) {
                dfs(isConnected, i, visited);
            }
        }
    }

    public int findCircleNum(int[][] isConnected) {
        int count = 0;
        boolean visited[] = new boolean[isConnected.length];
        for (int i = 0; i < isConnected.length; i++) {
            if (!visited[i]) {
                count++;
                dfs(isConnected, i, visited);
            }
        }
        return count;
    }
}
